Add documentation for RAID devices (#289)
NearNodeFlash/NearNodeFlash.github.io: docs/guides/storage-profiles/readme.md

In general, `scale` gives a simple way for users to get a filesystem that has performance consistent with their job size. `count` is useful for times when a user wants full control of the file system layout.

### RAID Configurations

> Allocations can be set up to use a RAID device to provide redundancy in the event of a drive failure. Optionally, commands can be specified to rebuild the RAID device after a replacement drive has been added. The storage profile parameters differ depending on whether the allocation is using LVM or zpool.

Reviewer: ```suggestion
Allocations can be set up to use a RAID device to provide continued access in the event of a drive failure. Optionally, commands can be specified to rebuild the RAID device after a replacement drive has been added. The storage profile parameters differ depending on whether the allocation is using LVM or zpool.
```

#### Zpool

> To create a Lustre confiuration with a redundant zpool, the `raidz` option should be added to the `zpoolCreate` command. To allow the zpool to be rebuilt with a new drive, the `zpoolReplace` command should be included.

Reviewer: ```suggestion
To create a Lustre confiuration with a redundant zpool, the `raidz` option is required in `zpoolCreate` command. To allow the zpool to be rebuilt with a new drive, the `zpoolReplace` command is required.
```

The example below shows a RAID configuration for the OST, however, the same options can be specified for any of the Lustre targets.

```yaml
apiVersion: nnf.cray.hpe.com/v1alpha8
kind: NnfStorageProfile
metadata:
  name: lustre-raid-example
  namespace: nnf-system
data:
[...]
  lustreStorage:
[...]
    ostCommandlines:
      mkfs: --ost --backfstype=$BACKFS --fsname=$FS_NAME --mgsnode=$MGS_NID --index=$INDEX
        --mkfsoptions="nnf:jobid=$JOBID" $ZVOL_NAME
      mountTarget: $ZVOL_NAME $MOUNT_PATH
      postActivate:
      - mountpoint $MOUNT_PATH
      zpoolCreate: -O canmount=off -o cachefile=none $POOL_NAME raidz $DEVICE_LIST
      zpoolReplace: $POOL_NAME $OLD_DEVICE $NEW_DEVICE
```

#### LVM

> A RAID logical volume can be used with XFS and Raw allocations. gfs2 allocations can not use RAID logical volumes because the LV is shared.

Reviewer: ```suggestion
A RAID logical volume can be used with XFS and Raw allocations. 
NOTE: gfs2 allocations cannot use RAID logical volumes because the LV is shared.
```

To create a redundant LV, `--type raid[x]` and `--nosync` should be specified in the `lvcreate` command. Also, the `--stripes` parameter should be adjusted accordingly to specify the number of data stripes. For `raid5`, `$DEVICE_NUM-1` is used.

> To allow the LV to be rebuilt after a replacement drive is added, `vgExtend`, `lvRepair`, and `vgReduce` should be specified in the `lvmRebuild` section.

Reviewer: ```suggestion
To allow the LV to rebuild after a drive is replaced, `vgExtend`, `lvRepair`, and `vgReduce` should be specified in the `lvmRebuild` section.
```
